src/include/uapi/ipc/topology.h

```diff
@@ -283,4 +284,16 @@ struct sof_ipc_pipe_comp_connect {
 	uint32_t sink_id;
 } __attribute__((packed));
 
+/* create new component kpb - SOF_IPC_TPLG_KPB_NEW */
+struct sof_ipc_comp_kpb {
+	struct sof_ipc_comp comp;
+	struct sof_ipc_comp_config config;
+	uint32_t size; /**< kpb size in bytes */
```

Comment:
This is really just a buffer. This pipeline only needs to use two buffers 
```
 PCM 0 <------------------------------+- Mux 0 <-- B0 <-- DMIC6 (DMIC01)
                                      |
 Keyword <-----B1 <---- Selector <----+
```  
So this component should be of type sof_ipc_comp_process (please align with @ranj063 for details) and it shoud be next to a buffer in the topology.

Reply:
@lgirdwood, as for now I don't see such type  *sof_ipc_comp_process*. This part needs to be clarified since I have no idea what parameters are going to be provided when KPB is created. Are things like sampling_freq, sampling_width, history_depth etc going to be included in that sof_ipc_comp_process structure which you mentioned?

Reply:
@lgirdwood I am not sure if Mux should be capable of all the logic that is being implemented inside the kpb right now (pls consider all the future scalability for multiple detectors being able to control their individual history depth etc.).
I believe that any future extensions of Mux should include current simple MicSel functionality etc. (in general being able to copy from multiple inputs to multiple outputs with the vector of channel coefficients provided for each input(s)-output set).